protocolport/protocol_port.go

```diff
@@ -0,0 +1,7 @@
+package protocolport
+
+type ProtocolPort struct {
+	IngressPort string `json:"ingressPort" yaml:"ingressPort"`
+	LBPort      string `json:"lbPort" yaml:"lbPort"`
+	Protocol    string `json:"protocol" yaml:"protocol"`
```

Comment:
I finished my creative work on the operator. Now I can deal with this here. IMO when you use a specific type you should do it to prevent others from making mistakes by using wrong values. IMO you cannot have a wrong value, since this setting is not used to define a real protocol. The `Protocol` property here is only used to create a service port name in the Kubernetes resources. Having `Protocol: "foo"`, the resulting service port name would be something like `foo-30010-al9qy`. See https://github.com/giantswarm/ingress-operator/pull/4/files#diff-a2f33643c72a67d64da15e2e20e88923R38. This name is maybe misguiding. I will also add godoc now to prevent such confusion. Let me know if you still want to use a dedicated type for this property. I would still strongly vote against it.